src/components/calloutForm.tsx

```diff
@@ -0,0 +1,186 @@
+}
+
+const calloutStyles = css`
+    border-top: 1px ${neutral[86]} solid;
+    border-bottom: 1px ${neutral[86]} solid;
+    position: relative;
+    margin: ${remSpace[4]} 0 ${remSpace[9]} 0;
+
+    &:not([open]) .is-on, &[open] .is-off {
+        display: none;
+    }
+
+    .is-on, .is-off {
+        position: absolute;
+        transform: translate(0, 50%);
+        bottom: 0;
+    }
+
+    &[open] {
+        background: ${neutral[97]};
+    }
+
+    ${darkModeCss`
+        background: white;
+        color: ${neutral[7]};
+        border: none;
+    `}
+`;
+
+const summaryStyles = css`
+    outline: none;
+    padding: 0;
+    list-style: none;
+
+    &::-webkit-details-marker {
+        display: none;
+    }
+`;
+
+const kickerStyles = css`
+    display: flex;
+    flex-direction: row;
+`;
+
+const logoStyles = css`
+    flex: initial;
+`;
+
+const headlineStyles = css`
+    ${headline.xxxsmall({ fontWeight: 'bold' })};
+    margin: 0;
+`;
+
+const descriptionStyles = css`
+    margin: ${remSpace[3]};
+
+    p {
+        ${body.small({ lineHeight: 'tight' })};
+        margin: ${remSpace[2]} 0;
+    }
+`;
+
+const errorStyles = css`
+    color: ${text.error};
+`;
+
+const speechBubbleStyles = (kicker: string): SerializedStyles => css`
+    padding: ${remSpace[3]};
+    position: relative;
+    color: ${neutral[100]};
+    background-color: ${kicker};
+    margin-bottom: ${remSpace[9]};
+    min-width: 5.5rem;
+
+    &::after {
+        content: '';
+        width: 1.25rem;
+        height: 1.375rem;
+        border-radius: 0 0 1.125rem;
+        position: absolute;
+        bottom: -0.75rem;
+        left: 0.625rem;
+        color: ${neutral[100]};
+        background-color: ${kicker};
+    }
+`;
+
+const formStyles = (kicker: string): SerializedStyles => css`
+    margin: ${remSpace[4]} ${remSpace[2]} ${remSpace[9]} ${remSpace[2]};
+
+    a {
+        color: ${kicker};
+        text-decoration: none;
+        ${ textSans.small() };
+        position: absolute;
+        bottom: ${remSpace[2]};
+        right: ${remSpace[2]};
+    }
+
+    button {
+        margin: ${remSpace[4]} 0;
+    }
+
+    input {
+        margin-bottom: ${remSpace[4]};
+    }
+
+    p {
+        ${textSans.small()};
+    }
+`;
+
+const renderField = ({ type, label, mandatory, options, id }: FormField): ReactElement | null => {
+    const fieldId = `field_${id}`;
+    switch (type) {
+        case 'text':
+            return <TextInput name={fieldId} label={label} optional={!mandatory} />
+        case 'textarea':
+            return <TextArea name={fieldId} label={label} optional={!mandatory} />
+        case 'file':
+            return <FileInput required={mandatory} name={fieldId} label={label} />
+        case 'radio':
+            return <RadioInput options={options} name={fieldId } label={label} />
+        default:
+            return null;
+    }
+}
+
+const CalloutForm: FC<CalloutProps> = (props: CalloutProps): ReactElement => {
```

Comment:
Just an FYI, I *think* `FC` provides a return type of `ReactElement | null` for you, so you could choose not to include the `: ReactElement` here.